### PR TITLE
Allow Native Broker by default

### PR DESCRIPTION
**Pull request by hectormmg**
This PR:

- Makes `system.allowNativeBroker` true by default
- Updates test suite to accommodate for new default behavior

### CODE REVIEW

File: change/@azure-msal-browser-f65d405e-9ee0-464a-92f9-19ce5ecaf595.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "major",
+  "comment": "Allow Native Broker by default #5616",
```
Comment (tnorling) [inline change/@azure-msal-browser-f65d405e-9ee0-464a-92f9-19ce5ecaf595.json]:
```suggestion
  "comment": "`allowNativeBroker` configuration defaults to `true` #5616",
```

**Review by tnorling — APPROVED** [review]

File: samples/msal-angular-v2-samples/angular14-rxjs7-sample-app/src/app/app.module.ts [hunk 2]
```diff
@@ -38,6 +38,7 @@ export function MSALInstanceFactory(): IPublicClientApplication {
       storeAuthStateInCookie: isIE, // set to true for IE 11. Remove this line to use Angular Universal
     },
     system: {
+      allowNativeBroker: false,
```
Comment (sameerag) [inline samples/msal-angular-v2-samples/angular14-rxjs7-sample-app/src/app/app.module.ts]: nit: Comments missing, looks like all .ts and .js files had comments: 
 
`//Disables WAM  Broker`

File: samples/msal-angular-v2-samples/angular14-rxjs6-sample-app/src/app/app.module.ts [hunk 3]
```diff
@@ -38,6 +38,7 @@ export function MSALInstanceFactory(): IPublicClientApplication {
       storeAuthStateInCookie: isIE, // set to true for IE 11. Remove this line to use Angular Universal
     },
     system: {
+      allowNativeBroker: false,
```
Comment (sameerag) [inline samples/msal-angular-v2-samples/angular14-rxjs6-sample-app/src/app/app.module.ts]: Same as the comment below: Comments missing.

**Review by sameerag — APPROVED** [review]
nit: Can we be consistent on commenting? Other than that lgtm.

One clarification: We are disabling to not disrupt the E2E samples, however, I assume eventually WAM sample is expected to be automated?? May be a discussion later, just want to call it out.

Approved.

**Comment by hectormmg** [comment]
> nit: Can we be consistent on commenting? Other than that lgtm.
> 
> One clarification: We are disabling to not disrupt the E2E samples, however, I assume eventually WAM sample is expected to be automated?? May be a discussion later, just want to call it out.
> 
> Approved.

I assume we'd ideally have E2E tests for it, but in this case disabling it on all other samples is just meant to keep the tests as valid as they were before the default change.

**Comment by sameerag** [comment]
> > nit: Can we be consistent on commenting? Other than that lgtm.
> > One clarification: We are disabling to not disrupt the E2E samples, however, I assume eventually WAM sample is expected to be automated?? May be a discussion later, just want to call it out.
> > Approved.
> 
> I assume we'd ideally have E2E tests for it, but in this case disabling it on all other samples is just meant to keep the tests as valid as they were before the default change.

I wonder if there is a better way to centralize the config. Way too many places to change. May be have our samples read from a env file??

**Comment by hectormmg** [comment]
> > > nit: Can we be consistent on commenting? Other than that lgtm.
> > > One clarification: We are disabling to not disrupt the E2E samples, however, I assume eventually WAM sample is expected to be automated?? May be a discussion later, just want to call it out.
> > > Approved.
> > 
> > 
> > I assume we'd ideally have E2E tests for it, but in this case disabling it on all other samples is just meant to keep the tests as valid as they were before the default change.
> 
> I wonder if there is a better way to centralize the config. Way too many places to change. May be have our samples read from a env file??

It might eliminate repetition for us but the configuration is an important part of the sample code in my opinion. Leaving it up to devs means we can't make PRs like this one where we make sure the sample still works after we change overrideable defaults.

File: lib/msal-browser/test/app/PCANonBrowser.spec.ts [hunk 4]
```diff
@@ -152,6 +152,9 @@ describe("Non-browser environment", () => {
         const instance = new PublicClientApplication({
             auth: {
                 clientId: TEST_CONFIG.MSAL_CLIENT_ID
+            },
+            system: {
+                allowNativeBroker: false
```
Comment (hectormmg) [inline lib/msal-browser/test/app/PCANonBrowser.spec.ts:157]:
```suggestion
                allowNativeBroker: false // Disables WAM Broker
```

File: samples/msal-angular-v2-samples/angular-b2c-sample-app/src/app/app.module.ts [hunk 5]
```diff
@@ -41,6 +41,7 @@ export function MSALInstanceFactory(): IPublicClientApplication {
       storeAuthStateInCookie: isIE, // set to true for IE 11
     },
     system: {
+      allowNativeBroker: false,
```
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular-b2c-sample-app/src/app/app.module.ts]:
```suggestion
      allowNativeBroker: false, // Disables WAM Broker
```

File: samples/msal-angular-v2-samples/angular10-sample-app/src/app/app.module.ts [hunk 6]
```diff
@@ -37,6 +37,7 @@ export function MSALInstanceFactory(): IPublicClientApplication {
       storeAuthStateInCookie: isIE, // set to true for IE 11
     },
     system: {
+      allowNativeBroker: false,
```
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular10-sample-app/src/app/app.module.ts]:
```suggestion
      allowNativeBroker: false, // Disables WAM Broker
```

File: samples/msal-angular-v2-samples/angular11-sample-app/src/app/app.module.ts [hunk 7]
```diff
@@ -39,6 +39,7 @@ export function MSALInstanceFactory(): IPublicClientApplication {
       storeAuthStateInCookie: isIE, // set to true for IE 11. Remove this line to use Angular Universal
     },
     system: {
+      allowNativeBroker: false,
```
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular11-sample-app/src/app/app.module.ts]:
```suggestion
      allowNativeBroker: false, // Disables WAM Broker
```

File: samples/msal-angular-v2-samples/angular12-sample-app/src/app/app.module.ts [hunk 8]
```diff
@@ -38,6 +38,7 @@ export function MSALInstanceFactory(): IPublicClientApplication {
       storeAuthStateInCookie: isIE, // set to true for IE 11. Remove this line to use Angular Universal
     },
     system: {
+      allowNativeBroker: false,
```
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular12-sample-app/src/app/app.module.ts]:
```suggestion
      allowNativeBroker: false, // Disables WAM Broker
```

File: samples/msal-angular-v2-samples/angular13-rxjs6-sample-app/src/app/app.module.ts [hunk 9]
```diff
@@ -38,6 +38,7 @@ export function MSALInstanceFactory(): IPublicClientApplication {
       storeAuthStateInCookie: isIE, // set to true for IE 11. Remove this line to use Angular Universal
     },
     system: {
+      allowNativeBroker: false,
```
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular13-rxjs6-sample-app/src/app/app.module.ts]:
```suggestion
      allowNativeBroker: false, // Disables WAM Broker
```

File: samples/msal-angular-v2-samples/angular13-rxjs7-sample-app/src/app/app.module.ts [hunk 10]
```diff
@@ -38,6 +38,7 @@ export function MSALInstanceFactory(): IPublicClientApplication {
       storeAuthStateInCookie: isIE, // set to true for IE 11. Remove this line to use Angular Universal
     },
     system: {
+      allowNativeBroker: false,
```
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular13-rxjs7-sample-app/src/app/app.module.ts]:
```suggestion
      allowNativeBroker: false, // Disables WAM Broker
```

File: samples/msal-angular-v2-samples/angular14-rxjs6-sample-app/src/app/app.module.ts (hunk 3 above)
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular14-rxjs6-sample-app/src/app/app.module.ts]:
```suggestion
      allowNativeBroker: false, // Disables WAM Broker
```

File: samples/msal-angular-v2-samples/angular14-rxjs7-sample-app/src/app/app.module.ts (hunk 2 above)
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular14-rxjs7-sample-app/src/app/app.module.ts]:
```suggestion
      allowNativeBroker: false, // Disables WAM Broker
```

File: samples/msal-angular-v2-samples/angular9-v2-sample-app/src/app/app.component.spec.ts [hunk 11]
```diff
@@ -9,6 +9,9 @@ function MSALInstanceFactory(): IPublicClientApplication {
     auth: {
       clientId: '6226576d-37e9-49eb-b201-ec1eeb0029b6',
       redirectUri: 'http://localhost:4200'
+    },
+    system: {
+      // allowNativeBroker: false // Disables WAM Bridge
```
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular9-v2-sample-app/src/app/app.component.spec.ts]:
```suggestion
     allowNativeBroker: false // Disables WAM Bridge
```

File: samples/msal-angular-v2-samples/angular9-v2-sample-app/src/app/app.module.ts [hunk 12]
```diff
@@ -38,6 +38,7 @@ export function MSALInstanceFactory(): IPublicClientApplication {
       storeAuthStateInCookie: isIE, // set to true for IE 11
     },
     system: {
+      allowNativeBroker: false,
```
Comment (hectormmg) [inline samples/msal-angular-v2-samples/angular9-v2-sample-app/src/app/app.module.ts]:
```suggestion
      allowNativeBroker: false, // Disables WAM Broker
```

File: samples/msal-browser-samples/VanillaJSTestApp2.0/app/navigateToLoginPage/authConfig.js [hunk 13]
```diff
@@ -11,6 +11,7 @@ const msalConfig = {
         storeAuthStateInCookie: false, // Set this to "true" if you are having issues on IE11 or Edge
     },
     system: {
+        allowNativeBroker: false,
```
Comment (hectormmg) [inline samples/msal-browser-samples/VanillaJSTestApp2.0/app/navigateToLoginPage/authConfig.js]:
```suggestion
        allowNativeBroker: false, // Disables WAM Broker
```

**Comment by sameerag** [comment]
> > > > nit: Can we be consistent on commenting? Other than that lgtm.
> > > > One clarification: We are disabling to not disrupt the E2E samples, however, I assume eventually WAM sample is expected to be automated?? May be a discussion later, just want to call it out.
> > > > Approved.
> > > 
> > > 
> > > I assume we'd ideally have E2E tests for it, but in this case disabling it on all other samples is just meant to keep the tests as valid as they were before the default change.
> > 
> > 
> > I wonder if there is a better way to centralize the config. Way too many places to change. May be have our samples read from a env file??
> 
> It might eliminate repetition for us but the configuration is an important part of the sample code in my opinion. Leaving it up to devs means we can't make PRs like this one where we make sure the sample still works after we change overrideable defaults.

I meant we devs will add the config code in the sample folder. Just once and loaded from a single source. However you may be right that this helps with the completeness of the code. Let us circle back on this later. 

Thanks.

**Pull request merged**